lsps2/store.go

```diff
@@ -3,22 +3,70 @@ package lsps2
 import (
 	"context"
 	"errors"
+	"log"
+	"time"
 
-	"github.com/breez/lspd/basetypes"
+	"github.com/breez/lspd/lightning"
+	"github.com/breez/lspd/lsps0"
 	"github.com/breez/lspd/shared"
+	"github.com/btcsuite/btcd/wire"
 )
 
+type SavePromises struct {
+	Menu  []*shared.OpeningFeeParams
+	Token string
+}
+
 type RegisterBuy struct {
 	LspId            string
 	PeerId           string
-	Scid             basetypes.ShortChannelID
+	Scid             lightning.ShortChannelID
+	OpeningFeeParams shared.OpeningFeeParams
+	PaymentSizeMsat  *uint64
+	Mode             OpeningMode
+}
+
+type BuyRegistration struct {
+	Id               uint64
+	LspId            string
+	PeerId           string // TODO: Make peerId in the registration a byte array.
+	Token            string
+	Scid             lightning.ShortChannelID
 	OpeningFeeParams shared.OpeningFeeParams
 	PaymentSizeMsat  *uint64
 	Mode             OpeningMode
+	ChannelPoint     *wire.OutPoint
+	IsComplete       bool
+}
+
+func (b *BuyRegistration) IsExpired() bool {
+	t, err := time.Parse(lsps0.TIME_FORMAT, b.OpeningFeeParams.ValidUntil)
+	if err != nil {
+		log.Printf("BuyRegistration.IsExpired(): time.Parse(%v, %v) error: %v", lsps0.TIME_FORMAT, b.OpeningFeeParams.ValidUntil, err)
+		return true
+	}
+
+	if time.Now().UTC().After(t) {
```

Comment:
NIT: Do we need the UTC conversion here?

Reply:
`ValidUntil` is utc, so in order to check for expiry, this needs UTC as well.